### ISSUE TITLE
Error: Corrupted compilation unit description

### ISSUE THREAD

**Issue by jvillard** [ocaml/dune#1157]
In dune 1.1.0 I non-deterministically observe "Error: Corrupted compilation unit description" when building in native mode. I haven't been able to reproduce the issue with 1.1.1so far so it may be that this is fixed already, but I thought I would flag anyway in case that's still interesting or someone searches for this error.

The opam switch is 4.06.1+flambda and the project is [infer](https://github.com/facebook/infer/).

Example run:

```console
$ dune build _build/default/infer.exe
[...]
    ocamlopt .InferModules.objs/InferModules__TaintTests.{cmx,o} (exit 2)
(cd _build/default && /home/jul/.opam/infer-4.06.1+flambda/bin/ocamlopt.opt -g -short-paths -safe-string -principal -strict-formats -strict-sequence -bin-annot -w +3+5+6+8+10+11+12+18+19+20+21+23+26+29+27+32+33+34+35+37+38+39+50+52+57+60-4-9-40-41-42-45-48 -open Core -open InferStdlib -open IStd -open InferGenerated )
File "TaintTests.ml", line 1:
Error: Corrupted compilation unit description
```

**Comment by diml** [comment]
Does TaintTests.ml have a .mli? If not, then this change could explain why it's fixed in 1.1.1:

> - `-opaque` wasn't correctly being added to modules without an interface.
  (#1108, fix #1107, @rgrinberg)

**Comment by jvillard** [comment]
Correct, thanks!
